### PR TITLE
update diligent-core tag signature

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **diligent-core/all**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
@AndreyMlashkin Could you please explain why does the checksum is no longer the same and point an upstream link related?

**Comment by AndreyMlashkin** [comment]
I have no idea
@Assiduous
@TheMostDiligent
I will try to ask maintainter

**Comment by TheMostDiligent** [comment]
I might have updated the tag, though this would be a month ago.

**Review by uilianries — DISMISSED** [review]

**Comment by uilianries** [comment]
@AndreyMlashkin Okay, it looks good, as the author confirmed the change. Still, some imports are outdated, due Conan 2.0. Could you please update them? The linter shows which imports need an update.

**Comment by AndreyMlashkin** [comment]
Sure, I can do that. But the whole conan2 linter changes seems to be find-and-replace thing - why it can't be automated?

**Comment by uilianries** [comment]
@AndreyMlashkin You can write a script to automate if you want ...

**Review by ericLemanissier — DISMISSED** [review]

**Comment by AndreyMlashkin** [comment]
> @AndreyMlashkin Okay, it looks good, as the author confirmed the change. Still, some imports are outdated, due Conan 2.0. Could you please update them? The linter shows which imports need an update.

I have tried to do it locally, full conan-v2 migration is quite complex for this recipe, like remove_files_by_mask is not available in conan v2.
I would postpone the migration

**Comment by AndreyMlashkin** [comment]
@prince-chrismc @SSE4, please approve

**Review by prince-chrismc — DISMISSED** [review]

**Comment by uilianries** [comment]
> I have tried to do it locally, full conan-v2 migration is quite complex for this recipe, like remove_files_by_mask is not available in conan v2.

@AndreyMlashkin remove_files_by_mask has been replaced by https://docs.conan.io/en/latest/reference/conanfile/tools/files/basic.html#conan-tools-files-rm

**Review by SSE4 — APPROVED** [review]

**Comment by AndreyMlashkin** [comment]
blocked by https://github.com/conan-io/conan-center-index/pull/12255

**Comment by AndreyMlashkin** [comment]
What is now a canonical way to delete PDB files?
this didn't work:
```rm(self, self.package_folder, "*.pdb")```

**Comment by AndreyMlashkin** [comment]
@uilianries @SSE4, what is a conan v2 way of deleting PDB files now?
this didn't work for me:
``` rm(self, self.package_folder, "*.pdb", recursive=True) ```

**Comment by prince-chrismc** [comment]
https://github.com/conan-io/conan-center-index/blob/d348c438f13c7e5828d8eb73cd8bf9c21f0cd712/docs/package_templates/cmake_package/all/conanfile.py#L140

**Comment by AndreyMlashkin** [comment]
@prince-chrismc still does not work....

File: recipes/diligent-core/all/conanfile.py [hunk 1]
```diff
@@ -151,37 +156,37 @@ def build(self):
...
-            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.dll")
+            rm(self, os.path.join(self.package_folder, "lib"), "*.dylib", recursive=True)
+            rm(self, os.path.join(self.package_folder, "lib"), "*.so", recursive=True)
+            rm(self, os.path.join(self.package_folder, "lib"), "*.dll", recursive=True)
 
         self.copy(pattern="*.fxh", dst="res", keep_path=False)
 
         self.copy("File2String*", src=os.path.join(self._build_subfolder, "bin"), dst="bin", keep_path=False)
-        tools.remove_files_by_mask(self.package_folder, "*.pdb")
+        rm(self, "*.pdb", self, os.path.join(self.package_folder, "lib"))
```
Comment (prince-chrismc) [inline recipes/diligent-core/all/conanfile.py:184]:
```suggestion
        rm(self, "*.pdb", os.path.join(self.package_folder, "lib"))
```

File: recipes/diligent-core/all/conanfile.py [hunk 2]
```diff
@@ -151,37 +156,37 @@ def build(self):
...
         # MinGw creates many invalid files, called objects.a, remove them here:
-        tools.remove_files_by_mask(self.package_folder, "objects.a")
+        rm(self, self.package_folder, "objects.a", recursive=True)
```
Comment (prince-chrismc) [inline recipes/diligent-core/all/conanfile.py:186]:
```suggestion
        rm(self, os.path.join(self.package_folder, "objects.a"), recursive=True)
```

**Comment by prince-chrismc** [comment]
The docs were bad https://github.com/conan-io/conan-center-index/pull/12987

File: recipes/diligent-core/all/conanfile.py (hunk 2 above)
Comment (SpaceIm) [reply recipes/diligent-core/all/conanfile.py:186]: No:

```python
rm(self, "objects.a", self.package_folder, recursive=True)
```

Comment (prince-chrismc) [reply recipes/diligent-core/all/conanfile.py:186]:
```suggestion
        rm(self, "objects.a", self.package_folder, recursive=True)
```

**Comment by AndreyMlashkin** [comment]
@SpaceIm @SSE4 @uilianries @prince-chrismc please approve

File: recipes/diligent-core/all/conanfile.py [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import cross_building, check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.files import rm, get, rmdir, rename, collect_libs, patches
+from conan.tools.apple import is_apple_os
 import os
-from conan.tools.build import cross_building
 
-required_conan_version = ">=1.33.0"
+required_conan_version = ">=1.50.0"
```
Comment (uilianries) [inline recipes/diligent-core/all/conanfile.py:10]:
```suggestion
required_conan_version = ">=1.51.3"
```
is_apple_os: https://docs.conan.io/en/latest/changelog.html#id30

File: recipes/diligent-core/all/conanfile.py [hunk 4]
```diff
@@ -83,14 +87,14 @@ def config_options(self):
...
-            tools.patch(**patch)
+        patches.apply_conandata_patches(self)
 
     def build_requirements(self):
-        self.build_requires("cmake/3.22.0")
+        self.tool_requires("cmake/3.24.0")
 
     def requirements(self):
         self.requires("opengl/system")
+        self.requires("wayland/1.21.0", override=True)
```
Comment (uilianries) [inline recipes/diligent-core/all/conanfile.py:97]:
```suggestion
        self.requires("wayland/1.21.0")
```
Not allowed. If have any conflict, it should be solved by aligning dependencies version

File: recipes/diligent-core/all/conanfile.py [hunk 5]
```diff
@@ -151,37 +155,37 @@ def build(self):
...
+        rmdir(self, os.path.join(self.package_folder, "lib"))
+        rmdir(self, os.path.join(self.package_folder, "bin"))
         self.copy("License.txt", dst="licenses", src=self._source_subfolder)
 
         if self.options.shared:
             self.copy(pattern="*.dylib", dst="lib", keep_path=False)
             self.copy(pattern="*.so", dst="lib", keep_path=False)
             self.copy(pattern="*.dll", dst="bin", keep_path=False)
-            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.a")
+            rm(self, os.path.join(self.package_folder, "lib"), "*.a", recursive=True)
```
Comment (uilianries) [inline recipes/diligent-core/all/conanfile.py:170]: It's True by default: https://docs.conan.io/en/latest/reference/conanfile/tools/files/basic.html#conan-tools-files-rm 

Of course, be explicit does not harm.

**Review by uilianries — COMMENTED** [review]
Please, remove override from require, it's not allowed.

**Comment by uilianries** [comment]
> @SpaceIm @SSE4 @uilianries @prince-chrismc please approve

Please, be patience, the CI bot ping us randomly, as soon as a PR passes.

**Comment by AndreyMlashkin** [comment]
> > @SpaceIm @SSE4 @uilianries @prince-chrismc please approve
> 
> Please, be patience, the CI bot ping us randomly, as soon as a PR passes.

This PR is an unexpected blocker for creating diligent-tools and diligent-fx and it lasts already more then a month. It should be only correction of a hash.

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by AndreyMlashkin** [comment]
@SpaceIm @SSE4 @prince-chrismc
Please take a look

**Review by SSE4 — APPROVED** [review]

**Pull request merged**